packages/autocomplete-core/onKeyDown.ts

```diff
@@ -112,6 +109,12 @@ export function onKeyDown<TItem>({
       suggestion.items[
         getRelativeHighlightedIndex({ state: store.getState(), suggestion })
       ];
+
+    if (item) {
+      // This prevents the `onSubmit` event to be sent when an item is selected.
+      event.preventDefault();
```

Comment:
When an item is highlighted, hitting "Enter" should select the item. If there's not highlighted item, it should trigger the submit event.